imnodes.cpp

```diff
@@ -1663,99 +1631,89 @@ void Shutdown(ImNodesContext* ctx) { EditorContextFree(ctx->DefaultEditorCtx); }
+static inline bool IsMiniMapActive() {
+    ImNodesEditorContext& editor = EditorContextGet();
+    return editor.MiniMapEnabled && editor.MiniMapSizeFraction > 0.0f;
+}
 
 static inline bool IsMiniMapHovered()
 {
+    ImNodesEditorContext& editor = EditorContextGet();
     return IsMiniMapActive() &&
            ImGui::IsMouseHoveringRect(
-               GImNodes->MiniMapRectScreenSpace.Min, GImNodes->MiniMapRectScreenSpace.Max);
-}
-
-static inline ImRect ToMiniMapRect(
-    const float                  minimap_size_fraction,
-    const ImRect&                editor_rect,
-    const ImNodesMiniMapLocation location)
-{
-    const ImVec2 editor_size(editor_rect.Max - editor_rect.Min);
-    const float  max_editor_coord = fmaxf(editor_size.x, editor_size.y);
-    const float  mini_map_coord = minimap_size_fraction * max_editor_coord;
-    const float  corner_offset_alpha = fminf(1.f - minimap_size_fraction, 0.1f);
-    const float  corner_offset_coord = corner_offset_alpha * mini_map_coord;
-
-    // Compute the size of the mini-map area; lower bound with some reasonable size values
-    const ImVec2 mini_map_size(mini_map_coord, mini_map_coord);
-
-    // Corner offset from editor context
-    const ImVec2 corner_offset(corner_offset_coord, corner_offset_coord);
-
-    switch (location)
-    {
-    case ImNodesMiniMapLocation_BottomRight:
-        return ImRect(
-            editor_rect.Max - corner_offset - mini_map_size, editor_rect.Max - corner_offset);
-    case ImNodesMiniMapLocation_BottomLeft:
-        return ImRect(
-            ImVec2(
-                editor_rect.Min.x + corner_offset.x,
-                editor_rect.Max.y - corner_offset.y - mini_map_size.y),
-            ImVec2(
-                editor_rect.Min.x + corner_offset.x + mini_map_size.x,
-                editor_rect.Max.y - corner_offset.y));
-    case ImNodesMiniMapLocation_TopRight:
-        return ImRect(
-            ImVec2(
-                editor_rect.Max.x - corner_offset.x - mini_map_size.x,
-                editor_rect.Min.y + corner_offset.y),
-            ImVec2(
-                editor_rect.Max.x - corner_offset.x,
-                editor_rect.Min.y + corner_offset.y + mini_map_size.y));
-    case ImNodesMiniMapLocation_TopLeft:
-        // [[fallthrough]]
-    default:
-        // [[fallthrough]]
-        break;
+               editor.MiniMapRectScreenSpace.Min, editor.MiniMapRectScreenSpace.Max);
+}
+
+static inline void CalcMiniMapLayout()
+{
+    ImNodesEditorContext& editor = EditorContextGet();
+    const ImVec2 offset = GImNodes->Style.MiniMapOffset;
+    const ImVec2 border = GImNodes->Style.MiniMapPadding;
+    const ImRect editor_rect = GImNodes->CanvasRectScreenSpace;
+
+    // Compute the size of the mini-map area
+    ImVec2 mini_map_size;
+    float mini_map_scaling;
+    {
+        const ImVec2 grid_content_size = ImFloor(editor.GridContentBounds.GetSize());
```

Comment:
It doesn't really matter, since this is used for content scaling of the mini-map, but since graph is empty mini-map has no content to draw. But I fixed it anyway, now it basically should result in 1.